examples/fluids/qfunctions/shocktube.h

```diff
@@ -0,0 +1,527 @@
+  // *INDENT-OFF*
+  // Inputs
+  const CeedScalar (*q)[CEED_Q_VLA] = (const CeedScalar(*)[CEED_Q_VLA])in[0],
+                   (*q_dot)[CEED_Q_VLA] = (const CeedScalar(*)[CEED_Q_VLA])in[2],
+                   (*q_data)[CEED_Q_VLA] = (const CeedScalar(*)[CEED_Q_VLA])in[3];
+  // Outputs
+  CeedScalar (*v)[CEED_Q_VLA] = (CeedScalar(*)[CEED_Q_VLA])out[0],
+             (*dv)[5][CEED_Q_VLA] = (CeedScalar(*)[5][CEED_Q_VLA])out[1];
+
+  const CeedScalar gamma     = 1.4;
+
+  CeedPragmaSIMD
+  // Quadrature Point Loop
+  for (CeedInt i=0; i<Q; i++) {
+    // *INDENT-OFF*
+    // Setup
+    // -- Interp in
+    const CeedScalar rho        =   q[0][i];
+    const CeedScalar u[3]       =  {q[1][i] / rho,
+                                    q[2][i] / rho,
+                                    q[3][i] / rho
+                                   };
+    const CeedScalar E          =   q[4][i];
+
+    // -- Interp-to-Interp q_data
+    const CeedScalar wdetJ      =   q_data[0][i];
+    // -- Interp-to-Grad q_data
+    // ---- Inverse of change of coordinate matrix: X_i,j
+    // *INDENT-OFF*
+    const CeedScalar dXdx[3][3] = {{q_data[1][i],
+                                    q_data[2][i],
+                                    q_data[3][i]},
+                                   {q_data[4][i],
+                                    q_data[5][i],
+                                    q_data[6][i]},
+                                   {q_data[7][i],
+                                    q_data[8][i],
+                                    q_data[9][i]}
+                                  };
+    // *INDENT-ON*
+    const CeedScalar
+    E_kinetic  = 0.5 * rho * (u[0]*u[0] + u[1]*u[1] + u[2]*u[2]),
+    E_internal = E - E_kinetic,
+    P          = E_internal * (gamma - 1); // P = pressure
+
+    // The Physics
+    // Zero v and dv so all future terms can safely sum into it
+    for (int j=0; j<5; j++) {
+      v[j][i] = 0;
+      for (int k=0; k<3; k++)
+        dv[k][j][i] = 0;
+    }
+    //-----mass matrix
+    for (int j=0; j<5; j++)
+      v[j][i] += wdetJ*q_dot[j][i];
+
+    // -- Density
+    // ---- u rho
+    for (int j=0; j<3; j++)
+      dv[j][0][i]  -= wdetJ*(rho*u[0]*dXdx[j][0] + rho*u[1]*dXdx[j][1] +
+                             rho*u[2]*dXdx[j][2]);
+    // -- Momentum
+    // ---- rho (u x u) + P I3
+    for (int j=0; j<3; j++)
+      for (int k=0; k<3; k++)
+        dv[k][j+1][i]  -= wdetJ*((rho*u[j]*u[0] + (j==0?P:0))*dXdx[k][0] +
+                                 (rho*u[j]*u[1] + (j==1?P:0))*dXdx[k][1] +
+                                 (rho*u[j]*u[2] + (j==2?P:0))*dXdx[k][2]);
+    // -- Total Energy Density
+    // ---- (E + P) u
+    for (int j=0; j<3; j++)
+      dv[j][4][i]  -= wdetJ * (E + P) * (u[0]*dXdx[j][0] + u[1]*dXdx[j][1] +
+                                         u[2]*dXdx[j][2]);
+  } // End Quadrature Point Loop
+
+  // Return
+  return 0;
+}
+// *****************************************************************************
+
```

Comment:
Do we need to duplicate this unstabilized Euler qfunction?

Reply:
Should I just remove this QFunction? Any addition of YZB into the implicit timestepping qfunction can occur in a generalized IFunction that will exist outside of `shocktube.h` and `eulervortex.h`?

Reply:
I went ahead and removed this.